src/SegmentButtonExpand/SegmentButtonExpand.tsx

```diff
@@ -1,31 +1,62 @@
 import cn from "classnames";
 import * as React from "react";
+import { FiltrationConditionComponentContext } from "../FiltrationConditionComponent/FiltrationConditionComponentContext";
 import { IconSvg } from "../IconSvg";
 import { SegmentButtonExpandProps as Props } from "./types";
 
 import "./SegmentButtonExpand.scss";
 
 export class SegmentButtonExpand extends React.Component<Props> {
+	public static context:
+		| ((children: React.ReactNode, filterAction: JSX.Element) => void)
+		| null;
+
+	public componentDidMount() {
+		const { isOpen, children } = this.props;
+		const renderPopover = this.context;
+
+		if (isOpen) {
+			renderPopover(children, this.filterAction());
+		}
+	}
+
+	public componentDidUpdate(prevProps: Props) {
+		const { isOpen, children } = this.props;
+		const renderPopover = this.context;
+
+		if (isOpen !== prevProps.isOpen) {
```

Comment:
Тут разве не надо и сам `isOpen` проверять?
```suggestion
		if (isOpen && isOpen !== prevProps.isOpen) {
```

Reply:
Или так
```suggestion
		if (isOpen && !prevProps.isOpen) {
```

Reply:
нет, не надо, с ним открытие/закрытие сегмента через происходит